statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,480 @@
+
+  * SurveyMean : Calculates the mean of each column
+
+  * SurveyTotal : Calculates the total of each column
+
+  * SurveyQuantile: Calculates the specified quantile[s] of each column
+"""
+
+import numpy as np 
+import pandas as pd 
+
+class SurveyDesign(object):
+    """                                                                                                                                              
+    Creates attributes that allow easy implementation of survey methods.                                                                                                   
+                                                                                                                                                     
+    Parameters                                                                                                                                          
+    -------                                                                                                                                          
+    strata : array-like or None                                                                                                                                 
+        Strata for each observation. If none, an array 
+        of ones is constructed                                                                                                                                 
+    cluster : array-like or None                                                                                                                 
+        Cluster for each observation. If none, an array 
+        of ones is constructed                                                                                                         
+    weights : array-like or None                                                                                                                               
+        The weight for each observation. If none, an array 
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    self.weights : (n, ) array
+        The weight for each observation
+    self.nstrat : integer
+        The number of district strata
+    self.sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    self.strat : (n, ) array
+        The related strata array from 0, 1, ...
+    self.ncs : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    self.sfclust : ndarray
+        The stratum for each cluster
+    self.nclust : integer
+        The total number of clusters across strata
+    """
+    
+    def __init__(self, strata=None, cluster=None, weights=None, nest=True):
+        strata, cluster, self.weights = self._check_args(strata, cluster, weights)
+        self.weights = weights
```

Comment:
This looks strange.  We set self.weights in the previous line then change it back here?